httpservice/src/HttpOutputStreamBuffer.h

```diff
@@ -54,7 +55,7 @@ class HttpOutputStreamBuffer : public std::streambuf
 
 private:
   std::vector<char> m_Buffer;
-  HttpServletResponsePrivate* m_Response;
+  std::shared_ptr<HttpServletResponsePrivate> m_Response;
```

Comment:
I think this will result in a smart-pointer life-cycle, leaking the HttpServletResponsePrivate object.

Reply:
I'm almost certain that this won't occur.

Reply:
I am afraid "almost" is not good enough :)

The `HttpServletResponsePrivate` instance can now hold on to a shared_ptr (instead of a raw pointer) to itself via the `m_HttpOutputStreamBuf` member. That is a classic smart pointer cycle and hence a memory leak in my book.

Reply:
Ahhh I missed this. I can convert it back.

Reply:
Ahhh I didn't see this. I will revert it back to using a raw pointer.